### PR TITLE
User Authentication

### PR DESCRIPTION
**Pull request by heidijane**
### **Changes**
- Set up an app controller, 3 page states are now tracked (register, login, and dashboard)
- Set up a dashboardLoader function that loads all components that require login
- User can now register an account
- If the user does not fill out a part of the registration form an error will be shown
- If the username is already registered, show an error
- If the e-mail is already in use, show an error
- If e-mail is not in a valid format, show an error
- If the passwords on the registration form do not match, an error will be shown
- User will automatically be logged in when they register
- User can now login
- If part of the form is not filled out an error will be shown
- If an incorrect username/password match is given, an error will be shown
- When the login info is verified a new session will be set to store the user's ID
- When the user logs in they will be taken to the dashboard and their username will be shown
- User can logout by clicking the logout button under their name
- added a login-register.css and imported it into main.css
- added the .hide class to main.css

### **Things to Test**
- That a user can register a new account
- That if the registration form is filled out incorrectly, errors are shown
- That the user should be automatically logged in after registering and should be taken directly to the dashboard
- That a user can login
- That the user must login using a valid username/password combo
- That the app state reflects the correct components
- That the correct user info appears on the dashboard even after logging out and logging back in several times
- That the user's dashboard will appear if the page is refreshed and the user is still logged in

### CODE REVIEW

**Review by iamdrewkirkland — APPROVED** [review]
Login/register features are functional and look good. Register form does not accept duplicate usernames or emails and presents error when incorrect. User login persists through refresh and logout button is functional. Changes look good!

File: src/index.html [hunk 1]
```diff
@@ -2,34 +2,38 @@
...
-  <meta charset="UTF-8">
-  <meta name="viewport" content="width=device-width, initial-scale=1.0">
-  <meta http-equiv="X-UA-Compatible" content="ie=edge">
-  <link rel="stylesheet" href="./styles/main.css">
-  <title>Nutshell Veridian Vicuna</title>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <link rel="stylesheet" href="./styles/main.css">
+    <title>Nutshell Veridian Vicuna</title>
```
Comment (midlantica) [inline src/index.html:9]: Does this work? Woohoo!

**Review by JaysonRice — APPROVED** [review]
All the components you mention look like they're functional!

**Review by midlantica — COMMENTED** [review]
Looks great!

**Pull request merged**